File: app/src/main/java/com/example/cryptoLearn/homeFragment.java
package com.example.cryptoLearn;

import android.content.Intent;
import android.graphics.Paint;
import android.graphics.Typeface;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.core.content.res.ResourcesCompat;
import androidx.fragment.app.Fragment;

import com.example.cryptoLearn.education.Categories;
import com.example.cryptoLearn.education.CategoryRepository;

import java.util.List;

public class homeFragment extends Fragment {

    private LinearLayout lessonCardContainer;
    private LinearLayout categoryNamesContainer;
    private CategoryRepository repository;
    private LinearLayout seeAllLayout;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_home, container, false);

        lessonCardContainer = view.findViewById(R.id.lesson_card_container);
        categoryNamesContainer = view.findViewById(R.id.category_names);
        seeAllLayout = view.findViewById(R.id.see_all);
        repository = CategoryRepository.getInstance();

        loadCategories();

        // Display lessons for the first category by default if categories are available
        List<String> categoriesList = repository.getCategories();
        if (!categoriesList.isEmpty()) {
            updateLessons(categoriesList.get(0));
        }

        seeAllLayout.setOnClickListener(v -> {
            Intent intent = new Intent(getActivity(), Categories.class);
            startActivity(intent);
        });
        return view;
    }

    private void loadCategories() {
        List<String> categoriesList = repository.getCategories();
        for (String category : categoriesList) {
            TextView categoryTextView = new TextView(requireContext());
            categoryTextView.setText(category);
            categoryTextView.setPadding(16, 8, 16, 8);
            categoryTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.black));
            categoryTextView.setTextSize(16);
            categoryTextView.setTypeface(ResourcesCompat.getFont(requireContext(), R.font.righteous));
            categoryTextView.setGravity(Gravity.CENTER);
            categoryTextView.setClickable(true);

            // Add click listener to update lessons
            categoryTextView.setOnClickListener(v -> updateLessons(category));

            // Add to the category container
            categoryNamesContainer.addView(categoryTextView);
        }
    }

    private void updateLessons(String category) {
        lessonCardContainer.removeAllViews();
        resetCategoryStyles();
        highlightSelectedCategory(category);

        List<CategoryRepository.Lesson> lessons = repository.getLessons(category);

        int marginTop8dp = getResources().getDimensionPixelSize(R.dimen.margin_top_8dp);
        Typeface righteousFont = ResourcesCompat.getFont(requireContext(), R.font.righteous);

        if (lessons != null) {
            for (int index = 0; index < lessons.size(); index++) {
                TextView lessonTextView = new TextView(requireContext());
                lessonTextView.setText(lessons.get(index).getTitle());
                lessonTextView.setPadding(40, 40, 40, 40);
                lessonTextView.setBackgroundResource(R.drawable.lesson_cards);
                lessonTextView.setCompoundDrawablesWithIntrinsicBounds(0, 0, R.drawable.check_outline, 0);
                lessonTextView.setCompoundDrawablePadding(8);
                lessonTextView.setTextSize(17f);
                lessonTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.black));
                lessonTextView.setTypeface(righteousFont);

                LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
                        LinearLayout.LayoutParams.MATCH_PARENT,
                        LinearLayout.LayoutParams.WRAP_CONTENT
                );
                if (index > 0) {
                    layoutParams.topMargin = marginTop8dp;
                }
                lessonTextView.setLayoutParams(layoutParams);

                lessonCardContainer.addView(lessonTextView);
            }
        }
    }

    private void resetCategoryStyles() {
        int defaultColor = ContextCompat.getColor(requireContext(), R.color.black);
        for (int i = 0; i < categoryNamesContainer.getChildCount(); i++) {
            TextView textView = (TextView) categoryNamesContainer.getChildAt(i);
            textView.setTypeface(Typeface.DEFAULT);
            textView.setPaintFlags(textView.getPaintFlags() & ~Paint.UNDERLINE_TEXT_FLAG);
            textView.setTextColor(defaultColor);
        }
    }

    private void highlightSelectedCategory(String category) {
        int highlightColor = ContextCompat.getColor(requireContext(), R.color.highlight);
        for (int i = 0; i < categoryNamesContainer.getChildCount(); i++) {
            TextView textView = (TextView) categoryNamesContainer.getChildAt(i);
            if (textView.getText().toString().equals(category)) {
                textView.setTypeface(textView.getTypeface(), Typeface.BOLD_ITALIC);
                textView.setPaintFlags(textView.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
                textView.setTextColor(highlightColor);
                break;
            }
        }
    }
}

File: app/src/main/java/com/example/cryptoLearn/homeFragment.kt
package com.example.cryptoLearn

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.core.content.res.ResourcesCompat
import androidx.fragment.app.Fragment
import com.example.cryptoLearn.R

class homeFragment : Fragment() {

    private lateinit var lessonCardContainer: LinearLayout
    private lateinit var categoryIntro: TextView
    private lateinit var categoryHistory: TextView
    private lateinit var categoryBlockchainNet: TextView

    // Define lesson content for each category
    private val lessonsMap = mapOf(
        "Introduction" to listOf("Lesson Name", "Lesson Name", "Lesson Name"),
        "History" to listOf("Lesson Name", "Lesson Name", "Lesson Name", "Lesson Name", "Lesson Name", "Lesson Name", "Lesson Name", "Lesson Name", "Lesson Name"),
        "Blockchain Networks" to listOf("Lesson Name", "Lesson Name", "Lesson Name")
    )

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_home, container, false)

        // Find views
        lessonCardContainer = view.findViewById(R.id.lesson_card_container)
        categoryIntro = view.findViewById(R.id.category_intro)
        categoryHistory = view.findViewById(R.id.category_history)
        categoryBlockchainNet = view.findViewById(R.id.category_blockchain_net)

        // Set click listeners for each category
        categoryIntro.setOnClickListener { updateLessons("Introduction", categoryIntro) }
        categoryHistory.setOnClickListener { updateLessons("History", categoryHistory) }
        categoryBlockchainNet.setOnClickListener { updateLessons("Blockchain Networks", categoryBlockchainNet) }

        updateLessons("Introduction", categoryIntro)


        return view
    }

    // Update the lesson section based on the selected category and style the selected category
    private fun updateLessons(category: String, selectedCategoryTextView: TextView) {
        // Clear previous lessons
        lessonCardContainer.removeAllViews()

        // Reset all categories to default style
        resetCategoryStyles()

        // Style the selected category
        selectedCategoryTextView.apply {
            setTypeface(typeface, android.graphics.Typeface.BOLD_ITALIC)
            paint.isUnderlineText = true
            setTextColor(ContextCompat.getColor(requireContext(), R.color.black))
        }

        // Get lessons for the selected category
        val lessons = lessonsMap[category]

        // Convert 8dp to pixels
        val marginTop8dp = resources.getDimensionPixelSize(R.dimen.margin_top_8dp)

        // Load the "Righteous" font from resources
        val righteousFont = ResourcesCompat.getFont(requireContext(), R.font.righteous)

        // Add each lesson as a TextView in the lesson container
        lessons?.forEachIndexed { index, lesson ->
            val lessonTextView = TextView(requireContext()).apply {
                text = lesson
                setPadding(40, 40, 40, 40)
                setBackgroundResource(R.drawable.lesson_cards)
                setCompoundDrawablesWithIntrinsicBounds(0, 0, R.drawable.check_outline, 0)
                compoundDrawablePadding = 8
                textSize = 17f
                setTextColor(ContextCompat.getColor(requireContext(), R.color.black))
                typeface = righteousFont // Apply the Righteous font
            }

            // Set top margin of 8dp for all items except the first one
            val layoutParams = LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
            )
            if (index > 0) {
                layoutParams.topMargin = marginTop8dp
            }
            lessonTextView.layoutParams = layoutParams

            // Add the TextView to the container
            lessonCardContainer.addView(lessonTextView)
        }
    }
    // Reset all categories to default style
    private fun resetCategoryStyles() {
        val defaultColor = ContextCompat.getColor(requireContext(), R.color.black)

        val categoryTextViews = listOf(categoryIntro, categoryHistory, categoryBlockchainNet)

        // Reset each category TextView to default styling
        categoryTextViews.forEach { textView ->
            textView.apply {
                typeface = android.graphics.Typeface.DEFAULT // Set to default typeface (normal, no bold or italic)
                paint.isUnderlineText = false
                setTextColor(defaultColor)
                invalidate()  // Force refresh to apply styling immediately
            }
        }
    }


}
